docs/DeveloperGuide.adoc

```diff
@@ -295,131 +304,434 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`*` |CS student with extra programmes| I want to verify that my study plan allows me to graduate with all the different requirements I have|
+|`*` |CS student with friends| I want to plan modules with my friends | we can take the same modules every semester
+|`*` |student who cares about grades| I want to be able to analyse my CAP per module, semester, year and overall| I can be more aware of my grades
+|`*` |user who admires beauty| I want to set each module with a color 1-8 | I can customise the look of my study plan
+|`*` |user who admires beauty| I want to change the color theme | I can customise the look of my study plan
+|`*` |experienced user who admires beauty| I want to change specific colors with hex code | I can customise the look of my study plan
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
-
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
 
-_{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `ModBuddy` and the *Actor* is the `Student`, unless specified otherwise)
 
-[discrete]
-=== Use case: Delete person
+=== Use case: UC01 - Create study plan
+
+*MSS*
+
+1. User chooses to create a study plan.
+2. User enters the requested description.
+3. ModBuddy displays the new study plan.
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. User chooses not to enter a description.
+** 2a1. ModBuddy creates a new study plan with a default description.
+** 2a2. Use case resumes from step 4.
+
++
+Use case ends.
+
+=== Use case: UC02 - Add module
+
+*MSS*
+
+1. Student requests to add a module to a particular semester.
+2. ModBuddy displays changes to study plan.
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. ModBuddy detects that the module entered does not exist.
+** 1a1. ModBuddy prompts Student to enter a correct module code
+** 1a2. User enters new module. 
++
+Steps 1a1-1a2 are repeated until the data entered are correct.
++
+Use case ends.
+
+=== Use case: UC03 - Commit a study plan
+
+*MSS*
+
+1. Student requests to save the current version of the study plan
+2. ModBuddy confirms that the version has been saved.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. ModBuddy detects that there have been no changes to the study plan from the previous commit.
+** 1a1. ModBuddy informs Student that the current version has already been saved.
++
+Use case ends.
+
+
+=== Use case: UC04 - Tag a module
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. Student requests to tag a module
```

Comment:
Can be specific and broken down into more steps e.g. Student selects a module, student tags the module